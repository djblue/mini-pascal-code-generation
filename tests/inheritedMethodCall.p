program main;

class thingy
begin
  var aa : integer;

  function setAA(ii : integer) : integer;
  begin
    aa := ii;
    setAA := aa
  end;

  function getAA : integer;
  begin
    getAA := aa
  end

end

class thing extends thingy
begin
  var bb : integer;
end

class main
begin

  function main;
    var aa : thing;
  begin
    aa := new thing;
    print aa.setAA(1024);
    aa.bb := 2048;
    print aa.getAA();
    print aa.bb;
    print aa.aa
  end

end
.
