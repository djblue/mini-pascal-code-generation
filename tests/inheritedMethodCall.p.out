1024
1024
2048
1024
